--- a/Makefile.in	2019-09-02 22:52:16.232394424 +0200
+++ b/Makefile.in	2019-09-02 22:53:40.199584250 +0200
@@ -241,8 +241,8 @@
 	chmod 755 ejabberd.init
 	#
 	# Service script
-	$(SED) -e "s*@ctlscriptpath@*$(SBINDIR)*g" ejabberd.service.template \
-		> ejabberd.service
+	[ -f ejabberd.service ] || $(SED) -e "s*@ctlscriptpath@*$(SBINDIR)*g" \
+		ejabberd.service.template > ejabberd.service
 	chmod 644 ejabberd.service
 	#
 	# Spool directory
