--- blender-2.79b-orig/intern/cycles/util/util_sseb.h	2018-03-24 02:22:25.000000000 +1100
+++ blender-2.79b/intern/cycles/util/util_sseb.h	2018-05-30 20:43:33.888717930 +1000
@@ -116,7 +116,7 @@
 __forceinline const sseb unpackhi( const sseb& a, const sseb& b ) { return _mm_unpackhi_ps(a, b); }
 
 template<size_t i0, size_t i1, size_t i2, size_t i3> __forceinline const sseb shuffle( const sseb& a ) {
-	return _mm_shuffle_epi32(a, _MM_SHUFFLE(i3, i2, i1, i0));
+	return _mm_castsi128_ps(_mm_shuffle_epi32(a, _MM_SHUFFLE(i3, i2, i1, i0)));
 }
 
 template<> __forceinline const sseb shuffle<0, 1, 0, 1>( const sseb& a ) {
--- blender-2.79b-orig/intern/itasc/kdl/tree.hpp	2018-03-24 02:22:25.000000000 +1100
+++ blender-2.79b/intern/itasc/kdl/tree.hpp	2018-05-30 20:33:52.045179988 +1000
@@ -34,7 +34,7 @@
     //Forward declaration
     class TreeElement;
     // Eigen allocator is needed for alignment of Eigen data types
-    typedef std::map<std::string,TreeElement, std::less<std::string>, Eigen::aligned_allocator<std::pair<std::string, TreeElement> > > SegmentMap;
+    typedef std::map<std::string,TreeElement, std::less<std::string>, Eigen::aligned_allocator<std::pair<const std::string, TreeElement> > > SegmentMap;
 
     class TreeElement
     {

--- blender-2.79b-orig/intern/cycles/device/device_cuda.cpp 2018-10-08 22:48:14.867737310 +0200
+++ blender-2.79b/intern/cycles/device/device_cuda.cpp 2018-10-08 22:49:49.903708562 +0200
@@ -334,7 +334,8 @@
 #ifdef WITH_CYCLES_DEBUG
 		cflags += " -D__KERNEL_DEBUG__";
 #endif
-
+		cflags += " -std=c++03";
+		
 		if(split) {
 			cflags += " -D__SPLIT__";
 		}
