diff --git a/cmake/FindOCC.cmake b/cmake/FindOCC.cmake
index 306bca6952..73f9af5d53 100644
--- a/cmake/FindOCC.cmake
+++ b/cmake/FindOCC.cmake
@@ -45,7 +45,7 @@ set( OCC_LIBS
     TKGeomAlgo
     TKGeomBase
     TKHLR
-    TKIGES
+    TKDEIGES
     TKLCAF
     TKMath
     TKMesh
@@ -56,18 +56,15 @@ set( OCC_LIBS
     TKRWMesh
     TKService
     TKShHealing
-    TKSTEP209
-    TKSTEPAttr
-    TKSTEPBase
-    TKSTEP
-    TKSTL
+    TKDESTEP
+    TKDESTL
     TKTObj
     TKTopAlgo
     TKV3d
-    TKVRML
+    TKDEVRML
     TKXCAF
-    TKXDEIGES
-    TKXDESTEP
+    TKXSDRAWIGES
+    TKXSDRAWSTEP
     TKXMesh
     TKXmlL
     TKXml
