--- a/tgp-msg.c
+++ b/tgp-msg.c
@@ -425,6 +425,11 @@ int tgp_msg_send (struct tgl_state *TLS, const char *message, tgl_peer_id_t to)
 }
 
 static char *tgp_msg_photo_display (struct tgl_state *TLS, const char *filename, int *flags) {
+  if (filename == NULL) {
+    *flags |= PURPLE_MESSAGE_SYSTEM;
+    return g_strdup_printf (_("[unavailable image]"));
+  }
+
   connection_data *conn = TLS->ev_base;
   int img = p2tgl_imgstore_add_with_id (filename);
   if (img <= 0) {
@@ -624,7 +629,7 @@ static void tgp_msg_display (struct tgl_state *TLS, struct tgp_msg_loading *C) {
 
       case tgl_message_media_photo:
         if (M->media.photo) {
-          g_return_if_fail(C->data != NULL);
+          // g_return_if_fail(C->data != NULL);
           text = tgp_msg_photo_display (TLS, C->data, &flags);
         }
         break;
@@ -1004,7 +1009,9 @@ void tgp_msg_recv (struct tgl_state *TLS, struct tgl_message *M, GList *before)
           // when fetching history. TODO: find out the reason for this behavior
           if (M->media.photo) {
             ++ C->pending;
-            tgl_do_load_photo (TLS, M->media.photo, tgp_msg_on_loaded_document, C);
+            // WORKAROUND for https://github.com/majn/telegram-purple/issues/517
+            // tgl_do_load_photo (TLS, M->media.photo, tgp_msg_on_loaded_document, C);
+            tgp_msg_on_loaded_document(TLS, C, TRUE, NULL);
           }
           break;
         }