--- a/fs/namei.c 2020-06-30 16:03:25.202995090 +0200
+++ b/fs/namei.c 2020-06-30 16:03:41.227670935 +0200
@@ -932,8 +932,8 @@
 		path_put(&last->link);
 }
 
-int sysctl_protected_symlinks __read_mostly = 0;
-int sysctl_protected_hardlinks __read_mostly = 0;
+int sysctl_protected_symlinks __read_mostly = 1;
+int sysctl_protected_hardlinks __read_mostly = 1;
 int sysctl_protected_fifos __read_mostly;
 int sysctl_protected_regular __read_mostly;
 
