diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 3091c2ca60df..f027bc7d2534 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -326,6 +326,8 @@
 	return ret;
 }
 
+EXPORT_SYMBOL(map_kernel_range);
+
 int is_vmalloc_or_module_addr(const void *x)
 {
 	/*
@@ -2130,6 +2132,8 @@
 				  GFP_KERNEL, caller);
 }
 
+EXPORT_SYMBOL(__get_vm_area_caller);
+
 /**
  * get_vm_area - reserve a contiguous kernel virtual area
  * @size:	 size of the area
