--- ./gegl/Makefile.am.ori	2020-01-15 23:03:33.812437241 +0100
+++ ./gegl/Makefile.am	2020-01-15 23:13:18.168414590 +0100
@@ -11,7 +11,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = \
     --warn-all \
-    --pkg="gegl-0.3" \
+    --pkg="gegl-0.4" \
     --pkg="glib-2.0" \
     --namespace="MyPaintGegl" \
     --nsversion="$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION)" \
@@ -37,10 +37,10 @@
 	../glib/mypaint-gegl-glib.c	\
 	mypaint-gegl-surface.c
 
-MyPaintGegl-@LIBMYPAINT_MAJOR_VERSION@.@LIBMYPAINT_MINOR_VERSION@.gir: libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la Makefile
-MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.3
+MyPaintGegl-@LIBMYPAINT_MAJOR_VERSION@.@LIBMYPAINT_MINOR_VERSION@.gir: libmypaint-gegl.la Makefile
+MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.4
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_CFLAGS = $(AM_CFLAGS) $(AM_CPPFLAGS) -I. -I..
-MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_LIBS = libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la ../libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la
+MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_LIBS = libmypaint-gegl.la ../libmypaint.la
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += MyPaintGegl-@LIBMYPAINT_MAJOR_VERSION@.@LIBMYPAINT_MINOR_VERSION@.gir
 
@@ -56,10 +56,10 @@
 
 ## pkg-config file ##
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.pc
+pkgconfig_DATA = libmypaint-gegl.pc
 
 ## libmypaint-gegl ##
-lib_LTLIBRARIES = libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la
+lib_LTLIBRARIES = libmypaint-gegl.la
 
 libmypaint_gegl_publicdir = $(includedir)/libmypaint-gegl
 
@@ -70,9 +70,9 @@
 	../glib/mypaint-gegl-glib.c	\
 	mypaint-gegl-surface.c
 
-libmypaint_gegl_@LIBMYPAINT_API_PLATFORM_VERSION@_la_SOURCES = $(libmypaint_gegl_public_HEADERS) $(LIBMYPAINT_GEGL_SOURCES)
+libmypaint_gegl_la_SOURCES = $(libmypaint_gegl_public_HEADERS) $(LIBMYPAINT_GEGL_SOURCES)
 
-libmypaint_gegl_@LIBMYPAINT_API_PLATFORM_VERSION@_la_CFLAGS = $(JSON_CFLAGS) $(GLIB_CFLAGS) $(GEGL_CFLAGS)
-libmypaint_gegl_@LIBMYPAINT_API_PLATFORM_VERSION@_la_LIBADD = $(top_builddir)/libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la $(GEGL_LIBS)
+libmypaint_gegl_la_CFLAGS = $(JSON_CFLAGS) $(GLIB_CFLAGS) $(GEGL_CFLAGS)
+libmypaint_gegl_la_LIBADD = $(top_builddir)/libmypaint.la $(GEGL_LIBS)
 
 endif # enable_gegl
--- ./gegl/mypaint-gegl-surface.c.ori	2020-01-15 23:05:52.896919345 +0100
+++ ./gegl/mypaint-gegl-surface.c	2020-01-15 23:07:52.792887407 +0100
@@ -78,7 +78,11 @@
 
     if (buffer_is_native(self)) {
         GeglBufferIterator *iterator = gegl_buffer_iterator_new(self->buffer, &tile_bbox, 0, self->format,
+        #if GEGL_MAJOR_VERSION == 0 && GEGL_MINOR_VERSION == 4 && GEGL_MICRO_VERSION >= 14
+                                      read_write_flags, GEGL_ABYSS_NONE, 8);
+        #else
                                       read_write_flags, GEGL_ABYSS_NONE);
+        #endif
 
         // Read out
         gboolean completed = gegl_buffer_iterator_next(iterator);
@@ -88,7 +92,11 @@
             g_critical("Unable to get tile aligned access to GeglBuffer");
             request->buffer = NULL;
         } else {
+            #if GEGL_MAJOR_VERSION == 0 && GEGL_MINOR_VERSION == 4 && GEGL_MICRO_VERSION >= 14
+            request->buffer = (uint16_t *)(iterator->items[0].data);
+            #else
             request->buffer = (uint16_t *)(iterator->data[0]);
+            #endif
         }
 
         // So we can finish the iterator in tile_request_end()
--- ./gegl/libmypaint-gegl.pc.in.ori	2020-01-15 23:04:55.419373035 +0100
+++ ./gegl/libmypaint-gegl.pc.in	2020-01-15 23:05:31.031472338 +0100
@@ -6,6 +6,6 @@
 Name: libmypaint
 Description: MyPaint brush engine library, with GEGL integration.
 Version: @LIBMYPAINT_VERSION@
-Requires: gegl-0.3 libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@
+Requires: gegl-0.4 libmypaint
 Cflags: -I${includedir}/libmypaint-gegl
 Libs: -L${libdir} -lmypaint-gegl
--- ./configure.ac.ori	2020-01-15 23:02:16.745386933 +0100
+++ ./configure.ac	2020-01-15 23:03:20.205781447 +0100
@@ -45,7 +45,7 @@
 
 ## Dependencies ##
 
-m4_define([gegl_required_version], [0.3])
+m4_define([gegl_required_version], [0.4.14])
 m4_define([introspection_required_version], [1.32.0])
 
 AC_INIT([libmypaint],
@@ -233,7 +233,7 @@
 
 if test "x$enable_i18n" != "xno"; then
   enable_i18n="yes"
-  GETTEXT_PACKAGE=libmypaint-libmypaint_api_platform_version
+  GETTEXT_PACKAGE=libmypaint
   AC_SUBST(GETTEXT_PACKAGE)
   AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE",
                      [The prefix for our gettext translation domains.])
@@ -272,7 +272,7 @@
 )
 
 if eval "test x$enable_gegl = xyes"; then
-  PKG_CHECK_MODULES(GEGL, gegl-0.3 >= gegl_required_version)
+  PKG_CHECK_MODULES(GEGL, gegl-0.4 >= gegl_required_version)
 fi
 AM_CONDITIONAL(ENABLE_GEGL, test "x$enable_gegl" = "xyes")
 
@@ -281,9 +281,9 @@
 
 AC_CONFIG_FILES([
   doc/Makefile
-  gegl/libmypaint-gegl-]libmypaint_api_platform_version()[.pc:gegl/libmypaint-gegl.pc.in
+  gegl/libmypaint-gegl.pc:gegl/libmypaint-gegl.pc.in
   gegl/Makefile
-  libmypaint-]libmypaint_api_platform_version()[.pc:libmypaint.pc.in
+  libmypaint.pc:libmypaint.pc.in
   Makefile
   po/Makefile.in
   tests/Makefile
--- ./Makefile.am.ori	2020-01-15 23:30:39.347381913 +0100
+++ ./Makefile.am	2020-01-15 23:32:34.895423478 +0100
@@ -58,10 +58,10 @@
 	tilemap.c
 
 # CAUTION: some of these need to use the underscored API version string.
-MyPaint-@LIBMYPAINT_API_PLATFORM_VERSION@.gir: libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la Makefile
+MyPaint-@LIBMYPAINT_API_PLATFORM_VERSION@.gir: libmypaint.la Makefile
 MyPaint_@LIBMYPAINT_API_PLATFORM_VERSION_UL@_gir_INCLUDES = GObject-2.0 GLib-2.0
 MyPaint_@LIBMYPAINT_API_PLATFORM_VERSION_UL@_gir_CFLAGS = $(AM_CFLAGS) $(AM_CPPFLAGS)
-MyPaint_@LIBMYPAINT_API_PLATFORM_VERSION_UL@_gir_LIBS = libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la
+MyPaint_@LIBMYPAINT_API_PLATFORM_VERSION_UL@_gir_LIBS = libmypaint.la
 MyPaint_@LIBMYPAINT_API_PLATFORM_VERSION_UL@_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += MyPaint-@LIBMYPAINT_API_PLATFORM_VERSION@.gir
 
@@ -79,7 +79,7 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 
-pkgconfig_DATA = libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.pc
+pkgconfig_DATA = libmypaint.pc
 
 ## libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@ ##
 
@@ -87,15 +87,15 @@
 
 LIBS = $(JSON_LIBS) $(GLIB_LIBS) @LIBS@
 
-lib_LTLIBRARIES = libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la
+lib_LTLIBRARIES = libmypaint.la
 
-libmypaint_@LIBMYPAINT_API_PLATFORM_VERSION@_la_LDFLAGS = \
+libmypaint_la_LDFLAGS = \
     -version-info @LIBMYPAINT_ABI_VERSION_INFO@ \
     -no-undefined
 
 #    -release @LIBMYPAINT_API_PLATFORM_VERSION@
 
-libmypaint_publicdir = $(includedir)/libmypaint-$(LIBMYPAINT_API_PLATFORM_VERSION)
+libmypaint_publicdir = $(includedir)/libmypaint
 
 nobase_libmypaint_public_HEADERS = \
 	mypaint-config.h				\
@@ -125,7 +125,7 @@
 	rng-double.c					\
 	tilemap.c
 
-libmypaint_@LIBMYPAINT_API_PLATFORM_VERSION@_la_SOURCES = $(libmypaint_public_HEADERS) $(LIBMYPAINT_SOURCES)
+libmypaint_la_SOURCES = $(libmypaint_public_HEADERS) $(LIBMYPAINT_SOURCES)
 
 DISTCLEANFILES = MyPaint-@LIBMYPAINT_API_PLATFORM_VERSION@.gir.files
 
--- ./tests/gegl/Makefile.am.ori	2020-01-15 23:34:57.606773858 +0100
+++ ./tests/gegl/Makefile.am	2020-01-15 23:35:17.369268775 +0100
@@ -29,8 +29,8 @@
 LDADD = \
 	$(DEPS)								\
 	$(top_builddir)/tests/libmypaint-tests.a \
-	$(top_builddir)/libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la		\
-	$(top_builddir)/gegl/libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la	\
+	$(top_builddir)/libmypaint.la		\
+	$(top_builddir)/gegl/libmypaint-gegl.la	\
 	$(GEGL_LIBS)
 
 endif
--- ./tests/Makefile.am.ori	2020-01-15 23:34:23.749639334 +0100
+++ ./tests/Makefile.am	2020-01-15 23:34:47.406034592 +0100
@@ -41,7 +41,7 @@
 LDADD = \
 	$(DEPS)							\
 	libmypaint-tests.a				\
-	$(top_builddir)/libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la
+	$(top_builddir)/libmypaint.la
 
 EXTRA_DIST = \
 	brushes/bulk.myb			\

--- ./libmypaint.pc.in.ori	2020-01-15 23:43:00.364450422 +0100
+++ ./libmypaint.pc.in	2020-01-15 23:43:29.303712397 +0100
@@ -8,5 +8,5 @@
 URL: @PACKAGE_URL@
 Version: @LIBMYPAINT_VERSION@
 Requires: @PKG_CONFIG_REQUIRES@
-Cflags: -I${includedir}/libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@
-Libs: -L${libdir} -lmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@ @OPENMP_CFLAGS@
+Cflags: -I${includedir}/libmypaint
+Libs: -L${libdir} -lmypaint @OPENMP_CFLAGS@
